### PR TITLE
Remove version and add status field to config objects

### PR DESCRIPTION
**Pull request by kyessenov**

### CODE REVIEW

File: model/registry.go [hunk 1]
```diff
@@ -14,110 +14,46 @@
...
 // ConfigKey is the identity of the configuration object
 type ConfigKey struct {
 	// Config object kind, e.g.  "MyKind"
 	Kind string
 	// Config object name, e.g. "my-name"
 	Name string
-	// Config version, e.g. "prod", "canary", "v1".
-	// "default" version is reserved, use ""
-	Version string
+	// Config object namespace, e.g. "default"
```
Comment (douglas-reid) [inline model/registry.go]: I might stick with convention here (start with export in comment).

Maybe something like: `// Namespace encodes the k8s-specific namespace of the ConfigKey, e.g. "default"`

File: model/validation.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	kindRegexp      = regexp.MustCompile("^[a-zA-Z][a-zA-Z0-9]*$")
+)
+
+// IsDNS1123Label tests for a string that conforms to the definition of a label in
+// DNS (RFC 1123).
+func IsDNS1123Label(value string) bool {
+	return len(value) <= dns1123LabelMaxLength && dns1123LabelRex.MatchString(value)
+}
+
+// Validate names in the config key
```
Comment (douglas-reid) [inline model/validation.go]: nit: `// Validate confirms that the names in the ConfigKey are appropriately formatted.`

File: model/validation.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	if !IsDNS1123Label(k.Name) {
+		return fmt.Errorf("Invalid name: %q", k.Name)
+	}
+	if !IsDNS1123Label(k.Namespace) {
+		return fmt.Errorf("Invalid namespace: %q", k.Name)
+	}
+	return nil
+}
+
+// Validate mapping
```
Comment (douglas-reid) [inline model/validation.go]: nit: `// Validate checks that each name in the KindMap conforms to spec and that the ProtoMessage type is valid.`

File: model/validation.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+			return fmt.Errorf("Invalid kind: %q", k)
+		}
+		if proto.MessageType(v.MessageName) == nil {
+			return fmt.Errorf("Cannot find proto message type: %q", v.MessageName)
+		}
+	}
+	return nil
+}
+
+// ValidateConfig object
```
Comment (douglas-reid) [inline model/validation.go]: nit: `// ValidateConfig ensures that a Config is valid by ...`

File: platform/kube/config.go [hunk 5]
```diff
@@ -27,9 +27,8 @@ import (
 type Config struct {
 	v1.TypeMeta `json:",inline"`
 	Metadata    api.ObjectMeta         `json:"metadata"`
-	Spec        map[string]interface{} `json:"data"`
-	// TODO: enable structured status field
-	Status string
+	Spec        map[string]interface{} `json:"spec"`
```
Comment (douglas-reid) [inline platform/kube/config.go]: worth doc (these are exported)?

File: model/registry.go [hunk 6]
```diff
@@ -14,110 +14,46 @@
...
 }
 
 // Config object holds the normalized config objects defined by Kind schema
 type Config struct {
 	ConfigKey
-	// Content holds the configuration struct
-	Content interface{}
+	// Spec holds the configuration struct
+	Spec interface{}
+	// Status holds the status information (may be null)
```
Comment (douglas-reid) [inline model/registry.go]: what sort of Status information do expect to be here?  Perhaps we could add some examples?  interface{} makes it hard to get a feel.

File: model/registry.go (hunk 1 above)
Comment (kyessenov) [inline model/registry.go]: Ack

File: model/validation.go (hunk 2 above)
Comment (kyessenov) [inline model/validation.go]: Ack

File: model/validation.go (hunk 3 above)
Comment (kyessenov) [inline model/validation.go]: Ack

File: model/validation.go (hunk 4 above)
Comment (kyessenov) [inline model/validation.go]: Ack

File: platform/kube/config.go (hunk 5 above)
Comment (kyessenov) [inline platform/kube/config.go]: They're same as the main declaration but exposed for Kubernetes serialization code.

**Pull request merged**